### ISSUE TITLE
Add linked reference to a program option through an intersphinx inventory file

### ISSUE THREAD

**Issue by eine** [sphinx-doc/sphinx#3487]
### Problem
I am trying to add external references to a documentation. References to sections are properly found and linked. However, references to options are found, but no link is created.

#### Procedure to reproduce the problem
- Add the following line to `conf.py`

```
intersphinx_mapping = {
	'ghdl':   ('http://ghdl-devfork.readthedocs.io/en/new-documentation-structure', None)
}
```
- Add these to the body of `index.rst`:
```
:option:`ghdl:-C`
:ref:`GHDL Roadmap <ghdl:CHANGE:Roadmap>`
```

#### Error logs / results
- http://poc-devfork.readthedocs.io/en/intertest/
#### Expected results
I expect the option to be a link to [`-C`](http://ghdl-devfork.readthedocs.io/en/new-documentation-structure/references/CommandReference.html#cmdoption-ghdl-mb-comments), since both urls are shown in the inventory file:
```
$ curl -s http://ghdl-devfork.readthedocs.io/en/new-documentation-structure/objects.inv | tail -n+5 | openssl zlib -d
...
-C std:option 1 references/CommandReference.html#cmdoption-ghdl-mb-comments -
...
change:roadmap std:label -1 changelog/Roadmap.html#change-roadmap Roadmap | Future improvements
...
```
### Reproducible project / your project
- https://github.com/1138-4EB/PoC/tree/intertest/docs

### Environment info
- OS: ReadTheDocs
- Python version: 3
- Sphinx version: 1.5.2

**Comment by tk0miya** [comment]
Thank you for reporting.
Fixed at 811df6b.

**Comment by Paebbels** [comment]
Does this also fix #2822?

**Comment by tk0miya** [comment]
Thanks!
